src/aleph/web/controllers/accounts.py

```diff
@@ -79,6 +83,56 @@ async def addresses_stats_view(request: web.Request):
     return web.json_response(output, dumps=lambda v: json.dumps(v))
 
 
+async def addresses_stats_view_v2(request: web.Request):
+    session_factory = get_session_factory_from_request(request)
+
+    try:
+        params = AddressesQueryParams.model_validate(request.query)
+    except ValidationError as e:
+        raise web.HTTPUnprocessableEntity(text=e.json())
+
+    with session_factory() as session:
+
+        # STEP 1 — Only search for addresses if the user provided a filter
+        if params.address_contains:
+            matched_addresses = find_matching_addresses(
+                session=session,
+                address_contains=params.address_contains,
+            )
+        else:
+            matched_addresses = None
+
+        # Fetch data and total count
+        stats, total_count = fetch_stats_for_addresses(
+            session=session,
+            addresses=matched_addresses,
+            sort_by=params.sort_by,
+            sort_order=params.sort_order,
+            filters=params.filters,
+            page=params.page,
+            per_page=params.pagination,
+        )
```

Comment:
Never do that directly, always filter the incoming values for the ones accepted, like variable types and so on. You have a lot of endpoints that receive message filters, please follow the same pattern.